core/errors.rs

```diff
@@ -3,72 +3,110 @@
 use rusty_v8 as v8;
 use std::any::Any;
 use std::any::TypeId;
+use std::borrow::Cow;
 use std::convert::TryFrom;
 use std::convert::TryInto;
 use std::error::Error;
 use std::fmt;
-use std::ops::Deref;
+use std::io;
 
 // The Send and Sync traits are required because deno is multithreaded and we
 // need to be able to handle errors across threads.
 pub trait AnyError: Any + Error + Send + Sync + 'static {}
 impl<T> AnyError for T where T: Any + Error + Send + Sync + Sized + 'static {}
 
 #[derive(Debug)]
-pub struct ErrBox(Box<dyn AnyError>);
+pub enum ErrBox {
+  Simple {
+    class: &'static str,
+    message: Cow<'static, str>,
+  },
+  Boxed(Box<dyn AnyError>),
```

Comment:
no class in the boxed case?

Reply:
The class is determined when it is needed. See `cli/errors.js`.

Reply:
In particular the function `rust_err_to_json()`